acados/sim/sim_gnsf.c

```diff
@@ -1715,9 +1716,10 @@ int sim_gnsf(void *config, sim_in *in, sim_out *out, void *args, void *mem_, voi
     int nxz2 = nx2 + nz2;
 
     // assert - only use supported features
-    if (mem->dt != in->T / opts->num_steps)
+    double epsilon = 1e-10 * fmax( fabs(mem->dt), fabs(in->T) );
```

Comment:
Did you get an issue with that?

Reply:
yes, the `sim_inverted_pendulum` example was failing on appveyor.
There, precompute is called after all options (and T) are set.

Reply:
https://randomascii.wordpress.com/2012/03/21/intermediate-floating-point-precision/

Reply:
oh i thought that the `DemoOfDanger` example applies here, but as dt and T are both `double` i am not so sure

Reply:
The use of the math library should be made with care. The function `fabs` is in the standard math library, but `fmax` has only been introduced with the c99 extension to the math library. So it should not be used, as generally not available in the math library for embedded devices.

On top of that, I would keep it simple, and just check that `abs(mem->dt - in->T/opts->num_steps) < 1e-12` or something like that. Leave a note that the chosen threshold `1e-12` is valid since all quantities are doubles, but should be changed if we (ever) make a single precision version.

Reply:
But getting back to @FreyJo 's comment, the quantity here should really be exactly equal, since `mem->dt` was computed in the first place using the exact same operation. Right?

Reply:
@FreyJo actually here the check only wants to catch the case of non-inizialisation. What about replacing this check on `dt` and instead introducing somewhere in the memory a bool/int flag which is set when everything is properly inizialised?

Reply:
@giaf thanks for pointing this out.
The check is there to verify that `mem->dt` (which is computed in precompute) is consistent with `T` and `num_steps`.
Since the latter ones can be always changed, i.e. also after the precompute step, we have to check for consistency.

Reply:
The problem was however caused by something else.
I reverted the change on gnsf and appveyor still passes.
I suggest to merge https://github.com/acados/acados/pull/446 instead.